keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/model/NamedImage.kt

```diff
@@ -34,7 +34,8 @@ val NamedImage.creationDate: Instant
     }
 
 val NamedImage.hasAppVersion: Boolean
-  get() = tagsByImageId
+  get() = tagsByImageId.values.isNotEmpty() &&
+    tagsByImageId
     .values
     .all { it != null && it.containsKey("appversion") }
```

Comment:
The problem was the vacuous case: `all` will return true if the collection is empty.